store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/retriever/impl/AccumuloElementsRetriever.java

```diff
@@ -18,18 +18,21 @@
 
 import uk.gov.gchq.gaffer.accumulostore.AccumuloStore;
 import uk.gov.gchq.gaffer.accumulostore.key.exception.IteratorSettingException;
-import uk.gov.gchq.gaffer.operation.SeedMatching;
+import uk.gov.gchq.gaffer.data.element.id.EdgeId;
 import uk.gov.gchq.gaffer.operation.impl.get.GetElements;
 import uk.gov.gchq.gaffer.store.StoreException;
 import uk.gov.gchq.gaffer.user.User;
 
+import java.util.stream.StreamSupport;
+
 public class AccumuloElementsRetriever extends AccumuloSingleIDRetriever<GetElements> {
     public AccumuloElementsRetriever(final AccumuloStore store,
                                      final GetElements operation,
                                      final User user)
             throws IteratorSettingException, StoreException {
         super(store, operation, user,
-                SeedMatching.SeedMatchingType.EQUAL != operation.getSeedMatching(),
+                // includeMatchedVertex if input only contains EntityIds
+                StreamSupport.stream(operation.getInput().spliterator(), false).noneMatch(input -> EdgeId.class.isInstance(input)),
```

Comment:
Potentially input could be large, but its a simple stream with an end-quick if any is matched.
IF your seed input is massive, you likely to have issues else where in gaffer.